### PR TITLE
Add more details to the serverless operator description

### PR DESCRIPTION
**Pull request by markito**
Update the description and include information about knative client. Includes a diagram for Knative Serving.

### CODE REVIEW

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -91,35 +91,96 @@ spec:
...
 
-    For documentation on using Knative Serving, see the
-    [serving section](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.3/html-single/serverless_applications/index#knative-serving).
+    For documentation on OpenShift Serverless, see the [installation
+    guide](https://docs.openshift.com/container-platform/4.3/serverless/installing-openshift-serverless.html)
+    or the [getting
+    started](https://docs.openshift.com/container-platform/4.3/serverless/serverless-getting-started.html)
+    sections.
+
+    Interested in learning even more ? Check out the [tutorials](https://redhat-developer-demos.github.io/knative-tutorial/knative-tutorial/index.html).
```
Comment (markusthoemmes) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]:
```suggestion
    Interested in learning even more? Check out the [tutorials](https://redhat-developer-demos.github.io/knative-tutorial/knative-tutorial/index.html).
```

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -91,35 +91,96 @@ spec:
...
+      revisions, traffic splits) 
+    - Growing support Knative eventing, closely following its development
+      (managing of sources & triggers) 
+    - A plugin architecture similar to that of kubectl plugins 
+    - A thin client-specific API in golang which helps in tasks like synchronously
+      waiting on Knative service write operations.  
+    - An easy integration of Knative into Tekton Pipelines by using kn in a Tekton
+      Task.
+
+    <!-- ### Knative Eventing
```
Comment (markusthoemmes) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: I'm not sure if the HTML comments work in here. I'd leave this out just in case.

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -91,35 +91,96 @@ spec:
...
+
+    The Serverless operator's provided APIs such as Knative Serving have certain
+    requirements with regards to the size of the underlying cluster. See [Getting
+    started with OpenShift
+    Serverless](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.3/html-single/serverless_applications/index#serverless-getting-started)
+    for more information.
+
+    # Supported Features
+
+    - **Easy to get started**: Provide a simplified developer experience to deploy
```
Comment (markusthoemmes) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: nit: AFAIK the colon should be bold too, typographically. https://writing.stackexchange.com/questions/14157/do-you-bold-punctuation-directly-after-bold-text

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -91,35 +91,96 @@ spec:
       version: v1alpha1
   minKubeVersion: 1.15.0
   description: |-
-    The Red Hat Serverless Operator provides a collection of API's to
-    install various "serverless" services.
+    The Red Hat OpenShift Serverless Operator provides a collection of APIs that
+    enables containers, microservices and functions to run "serverless".
+    Applications can scale up and down (to zero) on demand and be triggered by a
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Applications => Serverless applications

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -91,35 +91,96 @@ spec:
...
+    number of event sources. OpenShift Serverless integrates with a number of
+    platform services, such as Metering and Monitoring and it is based on the open
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: platform => OpenShift Container Platform

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -91,35 +91,96 @@ spec:
...
+    source project Knative.  
 
     This is a **[Tech Preview release](https://access.redhat.com/support/offerings/techpreview)!**
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Tech => Technology

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 7]
```diff
@@ -91,35 +91,96 @@ spec:
...
 
-    # Knative Serving
+    ## Prerequisites
+
+    The Serverless operator's provided APIs such as Knative Serving have certain
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Stay consistent with naming please :slightly_smiling_face: i.e. always use OpenShift Serverless

I'd rephrase this to make it easier to digest, maybe:

The APIs provided with the OpenShift Serverless operator require minimum cluster sizes on OpenShift Container Platform. For more information, see the documentation on [Getting started with OpenShift Serverless](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.3/html-single/serverless_applications/index#serverless-getting-started).

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 8]
```diff
@@ -91,35 +91,96 @@ spec:
...
+    requirements with regards to the size of the underlying cluster. See [Getting
+    started with OpenShift
+    Serverless](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.3/html-single/serverless_applications/index#serverless-getting-started)
+    for more information.
+
+    # Supported Features
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Keep headings same level?
i.e. why is pre-reqs ## but features is # ?

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml (hunk 3 above)
Comment (abrennan89) [reply olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: :100:

Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Provide => OpenShift Serverless provides

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 9]
```diff
@@ -91,35 +91,96 @@ spec:
...
+
+    - **Easy to get started**: Provide a simplified developer experience to deploy
+      and run cloud native applications on Kubernetes, providing powerful
+      abstractions.
+    - **Immutable Revisions**: Deploy new features performing canary, A/B or
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Are Immutable Revisions tested properly? I don't think we doc this anywhere except for a section on traffic splitting in the docs?

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 10]
```diff
@@ -91,35 +91,96 @@ spec:
...
+      blue-green testing with gradual traffic rollout following best practices.
+    - **Use any programming language or runtime of choice**: From Java, Python, Go
+      and JavaScript to Quarkus, SpringBoot or Node.js.
+    - **Automatic scaling:** No need to configure number of replicas, or idling.
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: - **Automatic scaling:** OpenShift Serverless removes the requirement to configure numbers of replicas or idling behavior. Applications automatically scale to zero when not in use, or scale up to meet demand, with built in reliability and fault tolerance.

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 11]
```diff
@@ -91,35 +91,96 @@ spec:
...
+      Scale to zero when not in use, auto scale to thousands during peak, with
+      built-in reliability and fault-tolerance.  
+    - **Event Driven Applications**: Build loosely coupled & distributed
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: - **Event Driven Applications:** You can build loosely coupled, distributed applications that can be connected to a variety of either built in or third party event sources, powered by operators.

Note - be consistent with either Operators or operators too - this YAML should be as consistent as possible with our actual docs. Don't add unnecessary dashes between words either, e.g. "third-party" => "third party"

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 12]
```diff
@@ -91,35 +91,96 @@ spec:
...
+      applications connecting with a variety of built-in or third-party event
+      sources connectors powered by Operators.  
+    - **Ready for the hybrid cloud**: True portable serverless running anywhere
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Don't use choppy sentences when writing - this is a doc, not a conversation. Do not use informal partial sentences.

- **Ready for the hybrid cloud:** OpenShift Serverless provides true, portable serverless functionality, that can run anywhere OpenShift Container Platform runs. You can leverage data locality and SaaS as you need it.

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 13]
```diff
@@ -91,35 +91,96 @@ spec:
...
+      OpenShift runs, that is on premise or on any cloud. Leverage data locality
+      and SaaS when needed.
 
-    Knative Serving builds on Kubernetes to support deploying and
-    serving of serverless applications and functions. Serving is easy
-    to get started with and scales to support advanced scenarios. The
-    Knative Serving project provides middleware primitives that
-    enable:
+    ## Components & APIs
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Again, be consistent with headings - some are #, some are ##, they should probably all be one or the other unless there's a structural reason why.

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 14]
```diff
@@ -91,35 +91,96 @@ spec:
...
+
+    Knative Serving builds on Kubernetes to support deploying and serving of
+    applications and functions as serverless containers. Serving is easy to get
+    started with and scales to support advanced scenarios. Other features
+    includes: 
 
     - Rapid deployment of serverless containers
     - Automatic scaling up and down to zero
     - Routing and network programming
     - Point-in-time snapshots of deployed code and configurations
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: It feels like we're just repeating information here.
I would just make this a bullet list e.g.

This operator provides the following components:

- Knative Serving
- Knative Client (`kn` CLI)
- Knative Eventing

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml [hunk 15]
```diff
@@ -91,35 +91,96 @@ spec:
...
 
     ## Further Information
 
-    For documentation on using Knative Serving, see the
-    [serving section](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.3/html-single/serverless_applications/index#knative-serving).
+    For documentation on OpenShift Serverless, see the [installation
+    guide](https://docs.openshift.com/container-platform/4.3/serverless/installing-openshift-serverless.html)
+    or the [getting
+    started](https://docs.openshift.com/container-platform/4.3/serverless/serverless-getting-started.html)
+    sections.
```
Comment (abrennan89) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: I don't think we need to include both, it just increases the chance of broken links and adds more maintenance. I would just use the getting started one.

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml (hunk 1 above)
Comment (abrennan89) [reply olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: Just my opinion, but I hate "Interested in learning even more?"
Instead, maybe say something like "If you would like to learn more about using OpenShift Serverless, try the [tutorials](https://redhat-developer-demos.github.io/knative-tutorial/knative-tutorial/index.html)."

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml (hunk 4 above)
Comment (markito) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]:
```suggestion
    Serverless applications can scale up and down (to zero) on demand and be triggered by a
```

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml (hunk 6 above)
Comment (markito) [inline olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]:
```suggestion
    This is a **[Technology Preview release](https://access.redhat.com/support/offerings/techpreview)!**
```

File: olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml (hunk 8 above)
Comment (markusthoemmes) [reply olm-catalog/serverless-operator/1.6.0/serverless-operator.v1.6.0.clusterserviceversion.yaml]: fwiw: Headings are not about importance but about hierarchy. You wouldn't have a top-level "##" heading.
